vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpacesCustomOrderManager.java

```diff
@@ -0,0 +1,157 @@
+    }
+
+    /**
+     * This method adds the newly created spaces to all custom orders at the end of the list
+     * @param space
+     */
+    @Override
+    public void addSpaceToCustomOrders(ISpace space) {
+        Iterable<SpacesCustomOrder> spacesCustomOrders = findAll();
+        for(ISpacesCustomOrder spaceCustomOrder :  spacesCustomOrders) {
+            List<ISpace> customOrderSpaces = spaceCustomOrder.getCustomOrderedSpaces();
+            if(!customOrderSpaces.contains(space)) {
+                spaceCustomOrder.getCustomOrderedSpaces().add(space);
+            }
+        }
+        save(spacesCustomOrders);
+    }
+    
+  
+
+    @Override
+    /**
+     * This method updates the custom order name and description
+     * @param spacesCustomOrderId
+     * @param title
+     * @param description
+     */
+    public void updateNameAndDescription(String spacesCustomOrderId, String title, String description) {
+        ISpacesCustomOrder spaceCustomOrder = get(spacesCustomOrderId);
+        spaceCustomOrder.setCustomOrderName(title);
+        spaceCustomOrder.setDescription(description);
+        spacesCustomOrderRepository.save((SpacesCustomOrder)spaceCustomOrder);
+    }
+
+    /**
+     * This method updated custom order of spaces
+     * @param spacesCustomOrderId
+     * @param spacesIds
+     * @return
+     */
+    @Override
+    public void updateSpaces(String spacesCustomOrderId, List<String> spacesIds) {
+        List<ISpace> spaces = new ArrayList<ISpace>();
+        spacesIds.stream()
+            .filter(spaceId -> spaceManager.getSpace(spaceId) != null)
+            .forEach(id -> spaces.add(spaceManager.getSpace(id)));
+  
+        ISpacesCustomOrder spaceCustomOrder = get(spacesCustomOrderId);
+        spaceCustomOrder.setCustomOrderedSpaces(spaces);
+        spacesCustomOrderRepository.save((SpacesCustomOrder)spaceCustomOrder);
+    }
+    
+    /**
+     * This method sets the custom order to be used on exhibition page
+     * @param customOrderId
+     */
+    @Override
+    public void setExhibitionSpacesCustomOrder(String customOrderId) {
+        IExhibition exhibition = exhibitionManager.getStartExhibition();
+        Optional<SpacesCustomOrder> spacesCustomOrder = spacesCustomOrderRepository
+                .findById(customOrderId);
+        if(spacesCustomOrder.isPresent())
+            exhibition.setSpacesCustomOrder(spacesCustomOrder.get());
+    }
+    
+    /**
+     * This method deletes the @{SpaceCustomOrder} object with the given id.
+     * @param id
+     */
+    @Override
+    public void delete(String id) {
+        IExhibition exhibition = exhibitionManager.getStartExhibition();
+        SpacesCustomOrder spaceCustomOrder = exhibition
+                .getSpacesCustomOrder();
+        if(spaceCustomOrder!=null && spaceCustomOrder.getId().equals(id)) {
+            exhibition.setSpacesCustomOrder(null);
+            exhibitionManager.storeExhibition((Exhibition)exhibition);
```

Comment:
the exhibition manager should not require an `Exhibition` object but an `IExhibition` object.